cmd/govim: watcher tests fails on macOS

<!-- Please answer these questions before submitting your issue. Thanks! -->

### What version of Vim/Gvim are you using?

<details><summary>Vim/Gvim version</summary><br><pre>
$ vim --version
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Jan 12 2020 10:35:50)
macOS version
Included patches: 1-111
Compiled by pontus@MacBook-Pro.localdomain
</pre></details>

### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.13.5 darwin/amd64
</pre>

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/pontus/Library/Caches/go-build"
GOENV="/Users/pontus/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/pontus/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/pontus/proj/govim/master/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/xm/ls208xd174v95pgd20rn_qbh0000gn/T/go-build360468964=/tmp/go-build -gno-record-gcc-switches -fno-common"
</pre></details>


### What version/commit of `govim` are you using?
Current master, 897d5a4

### What did you do?
`VIM_FLAVOR=vim go test .` in `cmd/govim`

### What did you expect to see?
No failures


### What did you see instead?

```
--- FAIL: TestScripts (10.46s)
    main_test.go:77: using gopls at "/var/folders/xm/ls208xd174v95pgd20rn_qbh0000gn/T/gobin-gopls-installdir034342552/gopls"
    --- FAIL: TestScripts/scenario_default (0.00s)
        --- FAIL: TestScripts/scenario_default/watcher (30.96s)
            testscript.go:348:
                # Test that the file watcher picks up file changes that occurs outside the editor
                # New file in the same package (29.583s)
                > vim ex 'e main.go'
                > cp const.go.orig const.go
                > errlogmatch '&protocol\.DidChangeWatchedFilesParams\{\n\S+:\s+Changes: \{\n\S+:\s+\{URI:"file://'$WORK/const.go'", Type:1\}'
                FAIL: testdata/scenario_default/watcher.txt:6: errlogmatch failed to find match

        --- FAIL: TestScripts/scenario_default/complete_watched (30.56s)
            testscript.go:348:
                # Test that ominfunc complete works where the completion is made
                # available in a file that is not loaded via the editor. (29.616s)
                > vim ex 'e main.go'
                > cp const.go.orig const.go
                > errlogmatch '&protocol\.DidChangeWatchedFilesParams\{\n\S+:\s+Changes: \{\n\S+:\s+\{URI:"file://'$WORK/const.go'", Type:1\}'
                FAIL: testdata/scenario_default/complete_watched.txt:6: errlogmatch failed to find match
```

--- a/cmd/govim/internal/fswatcher/os_darwin.go
+++ b/cmd/govim/internal/fswatcher/os_darwin.go
@@ -67,13 +67,18 @@ func New(gomodpath string, tomb *tomb.Tomb) (*FSWatcher, error) {
 			for i := range events {
 				event := events[i]
 				path := filepath.Join(mountPoint, event.Path)
+
+				// Darwin might include both "created" and "changed" in the same event
+				// so ordering matters below. The "created" case should be checked
+				// before "changed" to get a behavior that is more consistent with other
+				// os_other.go.
 				switch {
 				case event.Flags&fRemoved > 0:
 					eventCh <- Event{path, OpRemoved}
-				case event.Flags&fChanged > 0:
-					eventCh <- Event{path, OpChanged}
 				case event.Flags&fCreated > 0:
 					eventCh <- Event{path, OpCreated}
+				case event.Flags&fChanged > 0:
+					eventCh <- Event{path, OpChanged}
 				}
 			}
 		}

